cni/pkg/ambient/net.go

```diff
@@ -65,7 +65,7 @@ func AddPodToMesh(client kubernetes.Interface, pod *corev1.Pod, ip string) error
 }
 
 func addPodToMeshWithIptables(pod *corev1.Pod, ip string) error {
-	if ip == "" {
+	if ip == "" || ip == "<nil>" {
```

Comment:
under what case can it be nil

Reply:
+1, this seems like we may be handling it in the wrong place